tests/src/Gui/Camera.cpp

```diff
@@ -0,0 +1,220 @@
+#include <gtest/gtest.h>
+
+#include <array>
+#include <cmath>
+
+#include <Inventor/SbMatrix.h>
+#include <Inventor/SbRotation.h>
+#include <Inventor/SbViewVolume.h>
+#include <Base/Tools.h>
+#include <Gui/Camera.h>
+
+/*
+ This comment was previously used to get the hard coded axonometric view quaternions
+ in the Camera class.
+ This has since been replaced with unit tests that verify the correctness of the
+ quaternion calculations.
+
+ The old code is kept for reference and to show how the quaternions were calculated.
+
+ ---
+
+ Formulas to get quaternion for axonometric views:
+
+ \code
+from math import sqrt, degrees, asin, atan
+p1=App.Rotation(App.Vector(1,0,0),90)
+p2=App.Rotation(App.Vector(0,0,1),alpha)
+p3=App.Rotation(p2.multVec(App.Vector(1,0,0)),beta)
+p4=p3.multiply(p2).multiply(p1)
+
+from pivy import coin
+c=Gui.ActiveDocument.ActiveView.getCameraNode()
+c.orientation.setValue(*p4.Q)
+ \endcode
+
+ The angles alpha and beta depend on the type of axonometry
+ Isometric:
+ \code
+alpha=45
+beta=degrees(asin(-sqrt(1.0/3.0)))
+ \endcode
+
+ Dimetric:
+ \code
+alpha=degrees(asin(sqrt(1.0/8.0)))
+beta=degrees(-asin(1.0/3.0))
+ \endcode
+
+ Trimetric:
+ \code
+alpha=30.0
+beta=-35.0
+ \endcode
+
+ Verification code that the axonomtries are correct:
+
+ \code
+from pivy import coin
+c=Gui.ActiveDocument.ActiveView.getCameraNode()
+vo=App.Vector(c.getViewVolume().getMatrix().multVecMatrix(coin.SbVec3f(0,0,0)).getValue())
+vx=App.Vector(c.getViewVolume().getMatrix().multVecMatrix(coin.SbVec3f(10,0,0)).getValue())
+vy=App.Vector(c.getViewVolume().getMatrix().multVecMatrix(coin.SbVec3f(0,10,0)).getValue())
+vz=App.Vector(c.getViewVolume().getMatrix().multVecMatrix(coin.SbVec3f(0,0,10)).getValue())
+(vx-vo).Length
+(vy-vo).Length
+(vz-vo).Length
+
+# Projection
+vo.z=0
+vx.z=0
+vy.z=0
+vz.z=0
+
+(vx-vo).Length
+(vy-vo).Length
+(vz-vo).Length
+ \endcode
+
+ See also:
+ http://www.mathematik.uni-marburg.de/~thormae/lectures/graphics1/graphics_6_2_ger_web.html#1
+ http://www.mathematik.uni-marburg.de/~thormae/lectures/graphics1/code_v2/Axonometric/qt/Axonometric.cpp
+ https://de.wikipedia.org/wiki/Arkussinus_und_Arkuskosinus
+*/
+
+using namespace Base;
+
+namespace
+{
+
+Rotation buildAxonometricRotation(double alphaRad, double betaRad)
+{
+    const auto p1 = Rotation(Vector3d(1, 0, 0), Base::toRadians<float>(90.0));
+    const auto p2 = Rotation(Vector3d(0, 0, 1), alphaRad);
+    const auto p3 = Rotation(p2.multVec(Vector3d(1, 0, 0)), betaRad);
+    const auto p4 = p3 * p2 * p1;
+
+    return p4;
+}
+
+Base::Rotation toRotation(const SbRotation& sb)
+{
+    return Base::Rotation(sb[0], sb[1], sb[2], sb[3]);
+}
```

Comment:
Not needed, you can use `Base::convertTo<...>`. My suggestion would be to put `using Base::convertTo` on top.